packages/wallet/src/Wallets/BaseWallet.ts

```diff
@@ -551,7 +556,23 @@ export class BaseWallet implements ObservableWallet {
       utxo: this.utxo
     });
 
-    this.getPubDRepKey().catch(() => void 0);
+    const getPubDRepKey = async (): Promise<Ed25519PublicKeyHex | undefined> => {
+      if (isBip32PublicCredentialsManager(this.#publicCredentialsManager)) {
+        return (await this.#publicCredentialsManager.bip32Account.derivePublicKey(util.DREP_KEY_DERIVATION_PATH)).hex();
+      }
+
+      return undefined;
+    };
+
+    this.governance = {
+      getPubDRepKey,
+      isRegisteredAsDRep$: createDRepRegistrationTracker({
+        getPubDRepKey,
```

Comment:
@mkazlauskas could you please suggest if `getPubDRepKey` could be resolved before passing into the `createDRepRegistrationTracker` (so it accepts value but not func)? otherwise i need to resolve a promise in the `createDRepRegistrationTracker` and it breaks tests due to https://rxjs.dev/guide/testing/marble-testing#known-issues

Reply:
It can accept an observable, which you can create like `from(getPubDRepKey())`

Reply:
@mkazlauskas it is converted from promise like `from(getPubDRepKey())` , do you think it worth converting `getPubDRepKey` into observable within this pr?

Reply:
I think it's worth doing but not necessary for this PR, could be a separate refactor task some day :sweat_smile: